bitswap/client/internal/messagequeue/donthavetimeoutmgr.go

```diff
@@ -10,33 +10,41 @@ import (
 	"github.com/libp2p/go-libp2p/p2p/protocol/ping"
 )
 
-const (
-	// dontHaveTimeout is used to simulate a DONT_HAVE when communicating with
+func defaultDontHaveTimeoutConfig() *DontHaveTimeoutConfig {
+	cfg := &DontHaveTimeoutConfig{
+		DontHaveTimeout:            5 * time.Second,
+		MaxExpectedWantProcessTime: 2 * time.Second,
+		PingLatencyMultiplier:      3,
+		MessageLatencyAlpha:        0.5,
+		MessageLatencyMultiplier:   2,
+	}
+
+	cfg.MaxTimeout = cfg.DontHaveTimeout + cfg.MaxExpectedWantProcessTime
+	return cfg
```

Comment:
Two concerns with this approach:
1. We expose a way for overriding the defaults without exposing the implicit default values themselves
   -  people will keep hardcoding current values instead of referencing const that may change in the future
2. We use a struct which forces override fo all values
   - This makes it tedious to override a single value
   - This means if we add a new knob, people who already override the struct will override default with implicit zero value for that type (likely causing bugs)
  
My mental model for makign things like this configurable is to 
- have `WithFoo` configuration options which allow overriding each knob individually 
- and expose `DefaultFoo` to allow people to programmatically refer to implicit default, rather than hardcoding them in their own code.

This may be more work in this PR upfront, but will save us from headache in the future. Thoughts?

Reply:
I suggest we expose the default struct for people to override necessary knobs this way:
```go
cfg := DefaultDontHaveTimeoutConfig()
// example of granular customization
cfg.MaxExpectedWantProcessTime = 5 * time.Second
// use the global option.
WithDontHaveTimeoutConfig(cfg)
```

This solves your concern without creating "options hell". This is equivalent to having multi-level options, yet it's cleaner. Also, it doesn't require forwarding all the options from `client` to `bitswap` in case you want to add more options horizontally, which I tried initially, and got lost.

Reply:
Basically, the compromise I propose it expose the default configuration solving your concern and keeping things simple.